pkg/controller/che/che_controller.go

```diff
@@ -378,8 +384,25 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
+	if isOpenShift4 && instance.Spec.Auth.InitialOpenShiftOAuthUser != nil && !*instance.Spec.Auth.InitialOpenShiftOAuthUser && instance.Status.OpenShiftOAuthUserCredentialsSecret != "" {
+		if err := r.userHandler.DeleteOAuthInitialUser(deployContext); err != nil {
+			logrus.Errorf("Unable to delete initial OpenShift OAuth user from a cluster. Cause: %s", err.Error())
+			instance.Status.OpenShiftOAuthUserCredentialsSecret = ""
+			if err := r.UpdateCheCRStatus(instance, "openShiftOAuthUserCredentialsSecret", openShiftOAuthUserCredentialsSecret); err != nil {
+				return reconcile.Result{}, err
+			}
+			if err := r.UpdateCheCRSpec(instance, "InitialOpenShiftOAuthUser", "nil"); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		instance.Status.OpenShiftOAuthUserCredentialsSecret = ""
```

Comment:
nstance.Spec.Auth.OpenShiftoAuth = nil
instance.Spec.Auth.InitialOpenShiftOAuthUser = nil
and reconcile anyway